File: backend/models/message.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID

class MessageBase(BaseModel):
    receiver_id: UUID
    content: str
    is_encrypted: bool = False

class MessageCreate(MessageBase):
    pass

class MessageResponse(MessageBase):
    id: UUID
    sender_id: UUID
    created_at: datetime

    class Config:
        from_attributes = True

File: flutter_app/lib/models/message.dart
class Message {
  final String id;
  final String senderId;
  final String receiverId;
  final String content;
  final bool isPqcSigned;
  final String? signature;

  Message({
    required this.id,
    required this.senderId,
    required this.receiverId,
    required this.content,
    required this.isEncrypted,
    required this.createdAt,
    this.isPqcSigned = false,
    this.signature,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id'],
      senderId: json['sender_id'],
      receiverId: json['receiver_id'],
      content: json['content'],
      isEncrypted: json['is_encrypted'] ?? false,
      createdAt: DateTime.parse(json['created_at']),
      isPqcSigned: json['is_pqc_signed'] ?? false,
      signature: json['signature'],
    );
  }
}
